gui/dig.lua

```diff
@@ -553,20 +606,26 @@ function Dig:commit()
     -- Put any special logic for designation type here
     -- Right now it's setting the stair type based on the z-level
     function getDesignation(x, y, z)
+        local mode = self.subviews.mode_name:getOptionValue()
+        local stairs_type = self.subviews.stairs_subtype:getOptionValue()
 
         -- Nice stairs
-        if self.subviews.mode_name:getOptionValue() == "i" then
-            if math.abs(self:get_bounds().z1 - self:get_bounds().z2) == 0 then
-                return "`" -- return nothing, they need to specify more than one z-level
-            end
-            if z == 0 then
-                return "u" -- up stair
-            elseif z == math.abs(
-                self:get_bounds().z1 - self:get_bounds().z2
-            ) then
-                return "j" -- down stair
+        if mode == "i" then
+            if stairs_type == "auto" then
```

Comment:
if the player is extending down from an existing staircase, and the existing bottom layer is an up stair, then the top layer should become an UpDown instead of a Down. Similar for if they're extending an existing staircase upward (though I don't think you can convert a down stair into an up/down stair by digging..)

Reply:
This is a really good point. Vanilla does this with stairs so 'Auto' should match that